src/main/java/net/snowflake/client/core/SFArrowResultSet.java

```diff
@@ -277,9 +274,9 @@ private boolean fetchNextRowUnsorted()
 
           if (nextChunk == null)
           {
-            throw new SnowflakeSQLException(
+            throw new SnowflakeSQLLoggedException(
                 SqlState.INTERNAL_ERROR,
-                ErrorCode.INTERNAL_ERROR.getMessageCode(),
+                ErrorCode.INTERNAL_ERROR.getMessageCode(), session,
```

Comment:
Can you change the `session` argument to `Optional<SFSession>` for this constructor?

Reply:
The SFArrowResultSet constructor inherits the Session object from SFBaseResultSet and then initializes its session from the Statement object that ran the query corresponding to the ResultSet. 

Changing the type here would require changing several different ResultSet constructors in different classes and also refactoring all functions that use SFSession. It's a large-ish change so I'd prefer to have a separate PR for that.

Reply:
What I mean was to change the `session` argument for the `SnowflakeSQLLoggedException` constructor. I'm okay to leave it as is in this PR.